### PR TITLE
[Feature] Add conditional page refresh on specific routes and integrate `PayUI` component

### PR DESCRIPTION
**Pull request by jnsdls**
## PR-Codex overview
This PR updates components related to payment analytics and configuration. 

### Detailed summary
- Updated usePayPurchases, usePayVolume, usePayNewCustomers, and usePayCustomers hooks
- Added PayUI component for payment settings and analytics
- Updated DashboardConnectPayPage with new API key fetching logic

> The following files were skipped due to too many changes: `apps/dashboard/src/app/(dashboard)/dashboard/connect/pay/page.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3607?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3607** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3607?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3603** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3603?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3604](https://github.com/thirdweb-dev/js/pull/3604) <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3604?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jnsdls and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx [hunk 1]
```diff
@@ -89,10 +90,28 @@ export interface ConnectWalletProps {
   disableAddCustomNetwork?: boolean;
 }
 
+function useRefreshPageIfAppRouter() {
```
Comment (joaquim-verges) [inline apps/dashboard/src/@3rdweb-sdk/react/components/connect-wallet/index.tsx:93]: Oof, why do we need to refresh?

File: apps/dashboard/src/components/pay/PayAnalytics/hooks/usePayNewCustomers.ts [hunk 2]
```diff
@@ -30,7 +31,8 @@ export function usePayNewCustomers(options: {
   to: Date;
   intervalType: "day" | "week";
 }) {
-  const { user, isLoggedIn } = useLoggedInUser();
+  const { user } = useLoggedInUser();
+  const accessToken = useAccessToken();
 
   return useQuery(
     ["usePayNewCustomers", user?.address, options],
```
Comment (joaquim-verges) [inline apps/dashboard/src/components/pay/PayAnalytics/hooks/usePayNewCustomers.ts:38]: <i class='graphite__hidden'>[Re: line 41]<br/><br/></i>

Shouldn't this also go through a proxy?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3607?utm_source=unchanged-line-comment">Graphite</a>.</span>

Comment (jnsdls) [reply apps/dashboard/src/components/pay/PayAnalytics/hooks/usePayNewCustomers.ts:35]: does now :)

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**